server/backend/api/app/routers/router_guards.py

```diff
@@ -153,26 +163,33 @@ async def get_request_role_from_id(
     Some endpoints has path_params with "user_role_xref_id".
     Some endpoints has role_id in request body.
     """
-    user_role_xref_id = None
+    role = None
+
     if "user_role_xref_id" in request.path_params:
         user_role_xref_id = request.path_params["user_role_xref_id"]
```

Comment:
This whole section looks more complicated than I like, and makes certain assumptions about request parameters supplied on the routes this is guarding. I think at a minimum this needs better commenting of what the method is doing.